### PR TITLE
fix(type): function type in cjs require

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary

fix the function type in cjs require

close https://github.com/web-infra-dev/rspack/issues/8095

```ts
// in esm
import { rspack } from '@rspack/core';
rspack({ bail: true });

// in cjs
const rspack = require("@rspack/core");
// or const { rspack } = require("@rspack/core");
rspack({ bail: true });
```

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by SoonIter** [comment]
This is a solution with a little hack.
Maybe we need to ship both cjs and esm in the future.

File: packages/rspack-cli/tests/api/type/js-api-cjs.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import rspackCjsRequire, {
```
Comment (chenjiahan) [inline packages/rspack-cli/tests/api/type/js-api-cjs.test.ts:1]: What's the difference between this cjs case and the esm case? They seem to be the same.

Comment (SoonIter) [reply packages/rspack-cli/tests/api/type/js-api-cjs.test.ts:1]: one is mts, the other is ts

Comment (chenjiahan) [reply packages/rspack-cli/tests/api/type/js-api-cjs.test.ts:1]: get👌

**Comment by chenjiahan** [comment]
> Maybe we need to ship both cjs and esm in the future.

Agreed, `@rspack/core` should be a dual package.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**